zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java

```diff
@@ -9,18 +9,25 @@
 
 import io.camunda.zeebe.db.DbValue;
 import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.msgpack.property.ArrayProperty;
 import io.camunda.zeebe.msgpack.property.LongProperty;
 import io.camunda.zeebe.msgpack.property.StringProperty;
+import io.camunda.zeebe.msgpack.value.LongValue;
 import io.camunda.zeebe.util.buffer.BufferUtil;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
 
 public class PersistedGroup extends UnpackedObject implements DbValue {
 
   private final LongProperty groupKeyProp = new LongProperty("groupKey");
   private final StringProperty nameProp = new StringProperty("name");
+  private final ArrayProperty<LongValue> tenantKeysProp =
+      new ArrayProperty<>("tenantKeys", LongValue::new);
```

Comment:
❌ This should be aligned with users and mappings. The property should be a list of tenant ids, not keys. The reasoning behind this is that we must perform our tenant checks on ids. If we store the keys here that means we have to an additional state lookup for each of the tenants that is assigned to a group, which has an impact on performance. By storing a list of ids, we only need to get the group from the state and can perform cecks accordingly without looking up the tenant themselves.